src/plugins/intel_gpu/src/kernel_selector/kernels/activation/activation_kernel_opt.cpp

```diff
@@ -98,6 +98,10 @@ bool ActivationKernelOpt::Validate(const Params& p) const {
         }
     }
 
+    if (params.activations[0].function == ActivationFunction::SOFTPLUS && input_dt == Datatype::F16) {
```

Comment:
Perhaps an optimized kernel with CAN_USE_VECTOR=False will be faster than the reference one? Can you check?

Reply:
First of all, sorry for confusing you in the previous description. According to the updated description, `activation_opt` redefines the variable with the number 4, which always loads 4 elements at a time regardless of `vec_size`. It's hardcoded. In conclusion the best way to solve this problem is to use the ref kernel when `ActivationTypes `is `SoftPlus `and the data type is f16 at the same time.